process/test/ario.test.js

```diff
@@ -0,0 +1,423 @@
+const options = { format: "wasm64-unknown-emscripten-draft_2024_02_15" }
+
+// REFERENCE: https://github.com/ar-io/ar-io-mainnet-csvs # TODO - set this as a hash
+const EXPECTED_RECORD_COUNT = 2884
+const EXPECTED_VAULT_COUNT = 1446
+const EXPECTED_GATEWAY_COUNT = 277
+const EXPECTED_PRIMARY_NAME_COUNT = 575
+const EXPECTED_BALANCE_COUNT = 9162 // TODO: confirm why 9183 balances after 200 ticks
+const EXPECTED_TOTAL_SUPPLY = 10 ** 15; // 1B ARIO
+const EXPECTED_PROTOCOL_BALANCE = 65 * (10 ** 12); // 65M ARIO
+const env = {
+  Process: {
+    Id: 'AOS',
+    Owner: 'FOOBAR',
+    Tags: [
+      { name: 'Name', value: 'ARIO' },
+      { name: 'Authority', value: 'FOOBAR' }
+    ]
+  }
+}
+const handle = await AoLoader(wasm, options)
+const send = (memory, msg) => handle(memory, {
+    Target: 'AOS',
+    From: 'FOOBAR',
+    Owner: 'FOOBAR',
+    'Block-Height': '999',
+    Id: 'AOS',
+    Timestamp: '1740009600000', // 2025-02-20 00:00:00
+    Module: 'WOOPAWOOPA',
+    ...msg
+}, env)
+let start
+
+before(async () => {
+  // initialize initialize the wasm
+  const init = await send(null, {
+    Tags: [
+      { name: 'Name', value: 'ARIO' }
+    ]
+  })
+  // create the process
+  const msg = {
+    Tags: [
+      { name: 'Type', value: 'Process'}
+    ],
+  }
+  const createProcess = await send(init.Memory, msg)
+  // TODO: confirm this catches any failed module references or bad wasm errors
+  // assert there is no error log in create process
+  assert.ok(!createProcess.Output.data.includes('error'))
+  start = createProcess.Memory
+})
+
+test('perform a simple eval', async () => {
+ const msg = {
+    Tags: [
+      { name: 'Action', value: 'Eval' }
+    ],
+    Data: 'print("hello world")'
+  }
+  const result = await send(start, msg)
+  const output = result.Output.data
+  assert.ok(output.includes('"Message-Id": "'))
+  assert.ok(output.includes('"From": "FOOBAR"'))
+  assert.ok(output.includes('"Action": "Eval"'))
+  assert.ok(output.includes('"Timestamp": 1740009600000'))
+  assert.ok(output.includes('hello world'))
+})
+
+test('eval from non-owner address fails', async () => {
+  const msg = {
+    From: 'non-owner',
+    Owner: 'non-owner',
+    Tags: [
+      { name: 'Action', value: 'Eval' }
+    ],
+    Data: 'print("hello world")'
+  }
+  const result = await send(start, msg)
+  assert.ok(result.Messages.length === 0)
+})
+
+test('log an event on default handler getting called', async () => {
+  const result = await send(start, {
+    Tags: [
+      { name: 'Action', value: 'Not-An-Action' }
+    ],
+  })
+  const output = result.Output.data
+  assert.ok(output.includes('"Message-Id": "'))
+  assert.ok(output.includes('"From": "FOOBAR"'))
+  assert.ok(output.includes('"Default-Handler": true'))
+  assert.ok(output.includes('"Action": "Not-An-Action"'))
+  assert.ok(output.includes('"Timestamp": 1740009600000'))
+  assert.ok(result.Messages.length === 0)
+})
+
+test('return preloaded gateways', async () => {
```

Comment:
Anything we want to check around delegates here?

Reply:
The monitor tests assert that the delegate total stakes match the sum of all delegates staked on a given gateway